tests/integration_tests/test_osw_formatter_integration.py

```diff
@@ -38,46 +39,75 @@ def setUpClass(cls):
         formatter.start_listening.assert_called_once()
 
     @patch.object(OSWFomatterService, 'start_listening', new=MagicMock())
-    async def test_servicebus_receive(self):
+    def test_subscribe_to_upload_topic_with_pbf(self):
+        if os.environ['QUEUECONNECTION'] is None:
+            self.fail('QUEUECONNECTION environment not set')
+        formatter = OSWFomatterService()
+        self.test_data2['tdei_record_id'] = str(uuid.uuid4())
+        self.test_data2['tdei_project_group_id'] = str(uuid.uuid4())
+        upload_topic = self.core.get_topic(topic_name=self.upload_topic_name)
+        message = QueueMessage.data_from({'message': '', 'data': self.test_data2})
+        upload_topic.publish(data=message)
+        time.sleep(0.5)  # Wait to get the callback
+        formatter.start_listening.assert_called_once()
+
+    @patch.object(OSWFomatterService, 'start_listening', new=MagicMock())
+    async def test_servicebus_receive_with_zip(self):
         if os.environ['QUEUECONNECTION'] is None:
             self.fail('QUEUECONNECTION environment not set')
         formatter = OSWFomatterService()
         subscribe_function = MagicMock()
-        self.test_data['tdei_record_id'] = str(uuid.uuid4())
-        message = QueueMessage.data_from({'message': '', 'data': self.test_data})
+        self.test_data1['tdei_record_id'] = str(uuid.uuid4())
+        message = QueueMessage.data_from({'message': '', 'data': self.test_data1})
         formatter.publishing_topic.publish(data=message)
         validation_topic = self.core.get_topic(topic_name=self.validation_topic_name)
         async with validation_topic.subscribe(subscription='temp-validation-result', callback=subscribe_function):
             await asyncio.sleep(0.5)  # Wait for callback
             subscribe_function.assert_called_once()
             formatter.start_listening.assert_called_once()
 
-    def test_file_entity(self):
+    @patch.object(OSWFomatterService, 'start_listening', new=MagicMock())
+    async def test_servicebus_receive_with_pbf(self):
+        if os.environ['QUEUECONNECTION'] is None:
+            self.fail('QUEUECONNECTION environment not set')
+        formatter = OSWFomatterService()
+        subscribe_function = MagicMock()
+        self.test_data2['tdei_record_id'] = str(uuid.uuid4())
+        message = QueueMessage.data_from({'message': '', 'data': self.test_data2})
+        formatter.publishing_topic.publish(data=message)
+        validation_topic = self.core.get_topic(topic_name=self.validation_topic_name)
+        async with validation_topic.subscribe(subscription='temp-validation-result', callback=subscribe_function):
+            await asyncio.sleep(0.5)  # Wait for callback
+            subscribe_function.assert_called_once()
+            formatter.start_listening.assert_called_once()
+
+    def test_zip_file_entity(self):
         test_file_url = 'https://tdeisamplestorage.blob.core.windows.net/osw/test_upload/osw.zip'
```

Comment:
are these files always available? If not, it is better to keep these files in this repo as test/resources

Reply:
yes, its there